typing/shape_reduce.ml

```diff
@@ -271,6 +271,22 @@ end) = struct
     | NComp_unit s -> Comp_unit s
     | NError s -> Error s
 
+  (* When interested only of in the uids of aliased modules we do not read_back
+    the entire shape of the module, just enough to unroll the chain of aliases.
+  *)
+  let read_back_aliases_uids env (nf : nf) =
+    let force (Thunk (local_env, t)) =
+      reduce_ { env with local_env } t
+    in
```

Comment:
I think that `let force env (Trunk (local_env, t)) = ...` should be a function defined more globally, to be shared with the other copies of this definition.

Reply:
Done in [4c3f43f](https://github.com/ocaml/ocaml/pull/13001/commits/4c3f43faf46872fbe1315e8bd24b9d69a8f0090a)